### PR TITLE
Support Wagtail v2.14

### PR DESCRIPTION
**Pull request by fabienheureux**
No changes seem to be required from Wagtail 2.13 release notes: https://docs.wagtail.io/en/stable/releases/2.13.html#updated-handling-of-non-required-streamfields

However, I noticed that `stream_data` was deprecated in 2.12, whereas 2.12 is a valid requirement for Grapple.
I blindly replaced `stream_data` by `raw_data` as suggested in the 2.12 release notes (https://docs.wagtail.io/en/stable/releases/2.12.html), however I would like to have a second pair of eyes here. 

Tests pass at the moment.

### CODE REVIEW

**Comment by zerolab** [comment]
Thank you @fabienheureux

fwiw, `.stream_data` will be removed in 2.14 (https://github.com/wagtail/wagtail/blob/stable/2.12.x/wagtail/core/blocks/stream_block.py#L580)

I' would rather spend a bit more time and do this change properly  (as outlined in https://docs.wagtail.io/en/stable/releases/2.12.html#stream-data-on-streamfield-values-is-deprecated)

File: requirements.txt
Comment (ruisaraiva19) [inline requirements.txt:2]: Please follow the same convention as before:
```diff
-wagtail>=2.5,<=2.13
+wagtail>=2.5,<2.14
```

File: setup.cfg
Comment (ruisaraiva19) [inline setup.cfg:40]: Please follow the same convention as before:
```diff
-wagtail>=2.5, <=2.13
+wagtail>=2.5, <2.14
```

**Review by ruisaraiva19 — CHANGES_REQUESTED** [review]

File: grapple/types/streamfield.py
Comment (fabienheureux) [inline grapple/types/streamfield.py:25]: @zerolab I might lack a bit of knowledge regarding StreamField representation. 
Can this remain as is or should I update as well ?

Comment (zerolab) [reply grapple/types/streamfield.py:25]: `raw_data` is good.

Comment (zerolab) [reply grapple/types/streamfield.py:25]: saying that, this will fail for anything < 2.12 (and we have a `wagtail>=2.5, <2.14` constraint).

We want to support 2.11+, at the very least, so a good old try/except should do the trick here

Comment (fabienheureux) [reply grapple/types/streamfield.py:25]: Silly me 🤦 
Thanks for the review, I will update.

Comment (ruisaraiva19) [reply grapple/types/streamfield.py:25]: @fabienheureux you can now merge the latest code from the `main` branch which contains an updated CI pipeline that runs tests on multiple Wagtail versions (2.9, 2.10, 2.11, 2.12).

After merging you can update the [wagtail versions list](https://github.com/GrappleGQL/wagtail-grapple/blob/main/.github/workflows/ci.yml#L32) to have version 2.13 as well!

Comment (ruisaraiva19) [reply grapple/types/streamfield.py:25]: @fabienheureux let me know if you have any questions about the previous comment 😉

Comment (fabienheureux) [reply grapple/types/streamfield.py:25]: Thanks for the suggestion, no question it was perfectly clear (just lacked a bit of free time to work on this PR lately).

Comment (fabienheureux) [reply grapple/types/streamfield.py:25]: Test pass with all wagtail versions, I checked and it seems that `raw_data` property exists since a few Wagtail versions, for example here in 2.11 https://github.com/wagtail/wagtail/blob/1061caa5eff53f450520284b618e615fc2a809c8/wagtail/core/blocks/stream_block.py#L496

Comment (fabienheureux) [reply grapple/types/streamfield.py:25]: After a bit of debugging, I don't get where this code is used. 
Any idea ?

**Comment by fabienheureux** [comment]
I have a question about this conditionnal: https://github.com/GrappleGQL/wagtail-grapple/blob/main/grapple/types/streamfield.py#L111 

@zerolab do you have an idea of a query (in the example project) that could go through this conditionnal ? 
I am using this query 
```
{
  pages {
    ... on BlogPage {
      body {
        ...on StreamFieldInterface {
          rawValue
        }
      }
    }
  }
}
```

And can't go through. 
I am trying every streamfield block from the example project but I am not sure there is an existing block that matches this conditionnal.

I will continue digging this but maybe you have an idea.

**Comment by fabienheureux** [comment]
I use this PR on an actively developped project (wagtail 2.13) and I did not encounter any issues yet. 
Would it be possible to have a second pair of eyes regarding my question above ? 

Does anyone have a project, using wagtail < 2.13 on which it could be tested ?

File: .github/workflows/ci.yml
Comment (zerolab) [inline .github/workflows/ci.yml:32]:
```suggestion
        wagtail-version: ['2.11', '2.12', '2.13', '2.14']
```

File: grapple/types/streamfield.py
Comment (zerolab) [reply grapple/types/streamfield.py:25]:
```suggestion
        try
            return stream_value.raw_data
        except AttributeError:
            return stream_value.stream_data
```

File: requirements.txt
Comment (zerolab) [inline requirements.txt:2]:
```suggestion
wagtail>=2.11,<2.15
```

File: setup.cfg
Comment (zerolab) [inline setup.cfg:40]:
```suggestion
    wagtail>=2.11, <2.15
```

**Review by zerolab — CHANGES_REQUESTED** [review]
Hey @fabienheureux,

a couple of minor tweaks and we can get it in and do a release candidate

File: .github/workflows/ci.yml
Comment (fabienheureux) [reply .github/workflows/ci.yml:32]: This will require a bit more changes because there are compability issues with `wagtail_grapple[channels]` since django 3.0 is required for wagtail 2.14 (https://docs.wagtail.io/en/latest/releases/2.14.html) 

But I think it is worth the change, thanks for the suggestion

**Comment by fabienheureux** [comment]
> 
> 
> Hey @fabienheureux,
> 
> a couple of minor tweaks and we can get it in and do a release candidate

Thanks a lot for the suggestions !

**Review by ruisaraiva19 — APPROVED** [review]

**Pull request merged**
